scripts/generate.py

```diff
@@ -0,0 +1,492 @@
+        if "rootfs" in run.keys():
+            if fs == "initrd":
+                stream.write("    --initrd \"$rootfs\" \\\n")
+            if fs == "9pfs":
+                stream.write("    -v \"$rootfs\":/ \\\n")
+        if run["networking"]:
+            stream.write("    -a netdev.ipv4_addr=172.44.0.2 -a netdev.ipv4_gw_addr=172.44.0.1 \
+-a netdev.ipv4_subnet_mask=255.255.255.0 \\\n")
+        stream.write("    -- $cmd \\\n")
+
+        st = os.stat(out_file)
+        os.chmod(out_file, st.st_mode | stat.S_IEXEC)
+
+def generate_run():
+    """Generate running scripts.
+
+    Scripts are generated in scripts/run/ directory.
+    """
+
+    if not os.path.exists(RUN):
+        os.mkdir(RUN)
+
+    # Obtain configurations for running applications.
+    with open(os.path.join(SCRIPTS, "run.yaml"), "r", encoding="utf8") as stream:
+        data = yaml.safe_load(stream)
+        runs = data['runs']
+
+    # Obtain targets for basic QEMU and Firecracker-based runs from defconfigs.
+    def_targets = []
+    for file in files(DEFCONFIGS):
+        tokens = file.split('-')
+        if len(tokens) == 2:
+            item = (tokens[0], tokens[1], None)
+        elif len(tokens) == 3:
+            item = (tokens[0], tokens[1], tokens[2])
+        else:
+            print(f"Unknown tokens in deconfig file {file}")
+        def_targets.append(item)
+    def_targets = set(def_targets)
+
+    # Obtain targets for KraftKit runs form kraft.yaml.
+    with open(KRAFTCONFIG, "r", encoding="utf8") as stream:
+        data = yaml.safe_load(stream)
+        if 'targets' not in data.keys():
+            print(f"No targets in {KRAFTCONFIG}.")
+            return
+
+        # Walk through all configured applications.
+        for run in runs:
+            # Generate QEMU/Firecrakcer based scripts based on defconfigs.
+            for plat, arch, fs in def_targets:
+                if plat == "fc":
+                    generate_run_fc(run, arch, fs)
+                if plat == "qemu":
+                    generate_run_qemu(run, arch, fs)
+
+            # Generate KraftKit-based run scripts from kraft.yaml targets.
+            for target in data['targets']:
+                fs = None
+                if 'kconfig' in target:
+                    if "CONFIG_LIBVFSCORE_ROOTFS=\"9pfs\"" in target['kconfig']:
+                        fs = "9pfs"
+                    elif "CONFIG_LIBVFSCORE_ROOTFS=\"initrd\"" in target['kconfig']:
+                        fs = "initrd"
+                if 'name' in target.keys():
+                    name = target['name']
+                else:
+                    name = None
+                generate_run_kraft(run, name, target['architecture'], target['platform'], fs)
+
+def get_appname():
+    """Get the application name from the current directory.
+
+    Application name is the last part in the directory path.
+    """
+
+    global APPNAME
+
+    APPNAME = os.path.basename(os.getcwd())
```

Comment:
This might cause some problems, since we have the application name in the `defconfig` files, and it's not always the directory name. I don't know what is the better way to deal with this, maybe extract the APPNAME from the config UK_NAME.

Reply:
This won't work for KraftKit, as KraftKit doesn't rely on `defconfigs/`. The best solution is for users to have their directory named properly, as advised in the `README.md`. This will also not be a problem once applications are migrated to this repository (`app-testing`).